app/components/primer/alpha/segmented_control/button.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    class SegmentedControl
+      # `SegmentedControl::Button` is used inside `SegmentedControl` to render a button.
+      class Button < Primer::Component
+        status :alpha
+
+        ICON_ONLY_DEFAULT = false
+        ICON_ONLY_MAPPINGS = {
+          ICON_ONLY_DEFAULT => "",
+          true => "SegmentedControl-button--iconOnly",
+          :when_narrow => "SegmentedControl-button--iconOnly-whenNarrow",
+        }.freeze
```

Comment:
It's kinda weird to me to use both `true`/`false` and a symbol for keys here. What about something like

```suggestion
        ICON_ONLY_DEFAULT = :never
        ICON_ONLY_MAPPINGS = {
          ICON_ONLY_DEFAULT => "",
          :always => "SegmentedControl-button--iconOnly",
          :when_narrow => "SegmentedControl-button--iconOnly-whenNarrow",
        }.freeze
```